src/main/java/de/tum/in/www1/artemis/repository/specs/UserSpecs.java

```diff
@@ -0,0 +1,225 @@
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getInternalOrExternalSpecification(boolean internal, boolean external) {
+        if (!internal && !external) {
+            return null;
+        }
+        else {
+            return (root, query, criteriaBuilder) -> {
+                Predicate internalPredicate = criteriaBuilder.equal(root.get(User_.IS_INTERNAL), internal);
+                Predicate externalPredicate = criteriaBuilder.notEqual(root.get(User_.IS_INTERNAL), external);
+
+                return criteriaBuilder.and(internalPredicate, externalPredicate);
+            };
+        }
+    }
+
+    /**
+     * Creates the specification to match the state of the user (activated or deactivated).
+     *
+     * @param activated   true if the account should be activated
+     * @param deactivated true if the account should be deactivated
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getActivatedOrDeactivatedSpecification(boolean activated, boolean deactivated) {
+        if (!activated && !deactivated) {
+            return null;
+        }
+        else {
+            return (root, query, criteriaBuilder) -> {
+                Predicate activatedPredicate = criteriaBuilder.equal(root.get(User_.ACTIVATED), activated);
+                Predicate deactivatedPredicate = criteriaBuilder.notEqual(root.get(User_.ACTIVATED), deactivated);
+
+                return criteriaBuilder.and(activatedPredicate, deactivatedPredicate);
+            };
+        }
+    }
+
+    /**
+     * Creates the specification to match the users course.
+     *
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getAllUsersMatchingEmptyCourses() {
+        return (root, query, criteriaBuilder) -> criteriaBuilder.isEmpty(root.get(User_.GROUPS));
+    }
+
+    /**
+     * Creates the specification to match the users course.
+     *
+     * @param courseIds a set of courseIds which the users need to match
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getAllUsersMatchingCourses(Set<Long> courseIds) {
+        return (root, query, criteriaBuilder) -> {
+            Root<Course> courseRoot = query.from(Course.class);
+
+            Join<User, String> userToGroupJoin = root.join(User_.GROUPS, JoinType.LEFT);
+
+            updateAllUsersMatchingCoursesJoin(criteriaBuilder, courseRoot, userToGroupJoin);
+
+            query.groupBy(root.get(User_.ID)).having(criteriaBuilder.equal(criteriaBuilder.count(userToGroupJoin), courseIds.size()));
+
+            return criteriaBuilder.in(courseRoot.get(Course_.ID)).value(courseIds);
+        };
+    }
+
+    /**
+     * Helper method to update the course join. We join the users with the course via the groups of the users and the authority groups of the courses.
```

Comment:
```suggestion
     * Helper method to update the course join. We join the users with the course via the groups of the users and the authority groups of the courses.
     *
```
Formatting consistency with an empty line between comment and first `@param`